api/src/main/java/ai/djl/ndarray/gc/WeakReferenceWrapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.ndarray.gc;
+
+import java.lang.ref.ReferenceQueue;
+import java.lang.ref.WeakReference;
+
+/**
+ * {@code WeakReferenceWrapper<K, V>} extends a {@link WeakReference}. It uses an object of type K
+ * as the referent has an object property of type V.
+ */
+public class WeakReferenceWrapper<K, V> extends WeakReference<Object> {
```

Comment:
This class is unnecessary. You can keep it for now. We discuss after other issue is resolved.

Reply:
My idea was to use a standard `WeakHashMap` for storing the `PtNDArrayImpl` without altering it.
There is to my knowledge no hook in mechanism to use the internal `ReferenceQueue` of the `WeakHashMap` to get informed about no more referenced `PtNDArrayImpl`. Therefore I put a wrapper in front of the `WeakHashMap` with its own `ReferenceQueue`.